libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -200,6 +203,8 @@ class MapboxNavigation(
         }
     }
 
+    private val historyRecorderController = HistoryRecorderController(navigationOptions, logger)
```

Comment:
Would be great to provide this via `NavigationComponentProvider` so that we can test it.

Reply:
yeah going to find this is needed when updating MapboxNavigationTest

Reply:
ticketed because this wasn't entirely straight forward to handle the refresh test case

https://github.com/mapbox/mapbox-navigation-android/issues/4494